### ISSUE TITLE
Improve error message when a request is blocked (and maybe also if it fails in other ways)

### ISSUE THREAD

**Issue by flying-sheep** [jupyter/notebook#1957]
by naming a notebook “heatmap.ipynb”, I triggered an EasyList rule (`/heatmap.*?`) in uBlock, so a request to the following URL was apparently blocked: `/api/contents/heatmap.ipynb?type=notebook`

The end result was a nonsensical `Error loading Notebook: error`, which didn’t improve even after some fixes to error messages in master.

I’m filing this issue without all the misleading information in the original one: #1623

**Comment by flying-sheep** [comment]
To reproduce, install uBlock Origin and add a custom filter for `/blockednotebook.*?`, then try to open a notebook named `blockednotebook.ipynb`

**Comment by minrk** [comment]
Terrific, thanks for tracking down the cause, @flying-sheep!

**Comment by flying-sheep** [comment]
FYI: #2443 has been fixed, but the unhelpful error can still happen for other blocked names, so I recommend to leave this open.

**Comment by jasongrout** [comment]
I don't know that we can do anything about ublock blocking web resources - that's totally outside our control and in the user's hands. We do already suggest in the [documentation](https://jupyter-notebook.readthedocs.io/en/stable/troubleshooting.html#jupyter-doesn-t-load-or-doesn-t-work-in-the-browser):

> Try disabling any browser extensions and/or any Jupyter extensions you have installed.

Can you think of a way to surface that more visibly, or maybe expand on that in the documentation better?

**Comment by jbednar** [comment]
Sure, just make a more helpful error message than "Error loading notebook / Error". Even just making the message have some unique number or phrase would help, so that it becomes google-able, but ideally, if it's truly an error that Jupyter can say nothing about, then pointing to the documentation would be great. And if this is typically an error that happens with ad blockers, pointing to that specific section of the docs would be super helpful!

**Comment by flying-sheep** [comment]
Yeah, exactly that’s what I proposed in the title and original support. My proposal is

1. show a generic failure message when API requests fail, including path and status
2. in case a request to fetch notebook contents fail, add a line about possible ad blockers interfering

> API request to /api/contents/heatmap.ipynb?type=notebook failed with HTTP status 404: not found
> 
> This failure indicates that an ad blocker might block your notebook’s name.
